### ISSUE TITLE
Regent: error on attempting to access field of type int3d

### ISSUE THREAD

**Issue by elliottslaughter** [StanfordLegion/legion#317]
Submitting this on behalf of @manopapad to track a bug in the Regent compiler. This very simple example fails with a compile error:

```
import "regent"
struct particles_columns {
  cell : int3d;
}
task main()
  var v = int3d{0,0,0}
  v.x = 12
  var is = ispace(int1d, 10)
  var r1 = region(is, particles_columns)
  for v in r1 do
    v.cell.x = 0
  end
end
regentlib.start(main)
```

Fails with error:

```
no_x_in_int3d.rg:15: no field 'x' in type int3d
   v.cell.x = 0
    ^
```

**Comment by elliottslaughter** [comment]
Spent a little more time looking at this today. I think this got broken when we moved to deppart and `int*d` types got turned into a indivisible field (previously they had been separate fields). The basic problem is that from the expression `v.cell` is returning the basic unit of the field, from the perspective of the field accessor. If you want to do `v.cell.x` (especially as an l-value) you then are accessing some offset within the field. I don't think the code generator is smart enough to realize this at the moment, so even if you fix it at the type checker level further fixes are needed for codegen.

Adding @magnatelee since he was the one who did the original conversion to indivisible fields.

**Comment by magnatelee** [comment]
This should be fixed as of b93d076c. @elliottslaughter Could you confirm the fix and close this issue?

**Comment by elliottslaughter** [comment]
Fixed, thanks!
